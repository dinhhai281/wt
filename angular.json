{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "angular-unit-test": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "changeDetection": "OnPush",
          "viewEncapsulation": "None"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "outputPath": "dist/angular-unit-test",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "aot": true,
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss"
            ],
            "scripts": [],
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-builders/custom-webpack:dev-server",
          "options": {
            "browserTarget": "angular-unit-test:build",
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "browserTarget": "angular-unit-test:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "angular-unit-test:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "src/**/*.ts",
              "src/**/*.component.html"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "angular-unit-test"
}
