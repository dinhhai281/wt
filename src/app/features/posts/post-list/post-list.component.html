<div class="grid grid-flow-row">
  <article
    class="border-red-500 border-2 rounded-md bg-white p-4 mb-2 grid grid-cols-4 gap-2 cursor-pointer"
    *ngFor="let post of posts"
    (click)="select$.next(post.id)"
  >
    <div class="col-span-1">
      {{ post.id }}
      {{ post.userId }}
    </div>
    <div class="col-span-3">
      <div class="text-lg text-pink-700 font-semibold">
        {{ post.title }}
      </div>
      <div>
        {{ post.body }}
      </div>
    </div>
  </article>
</div>
