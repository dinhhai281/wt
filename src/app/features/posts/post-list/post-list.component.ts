import {
  Component,
  OnInit,
  ViewEncapsulation,
  ChangeDetectionStrategy,
  Input,
  Output,
  EventEmitter,
} from '@angular/core';
import { Post } from '@app/models';

@Component({
  selector: 'app-post-list',
  templateUrl: './post-list.component.html',
  styleUrls: ['./post-list.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PostListComponent implements OnInit {
  @Input() posts: Post[];
  @Output('selectPost') select$ = new EventEmitter<number>();

  constructor() {}

  ngOnInit(): void {}
}
