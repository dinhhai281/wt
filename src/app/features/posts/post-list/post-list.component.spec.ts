import { render, RenderResult } from '@testing-library/angular';

import { PostListComponent } from './post-list.component';

describe('PostListComponent', () => {
  let component: RenderResult<PostListComponent>;

  beforeEach(async () => {
    component = await render(PostListComponent, {
      componentProperties: {
        posts: [{ title: 'title', id: 1, userId: 1, body: 'body' }],
      },
    });
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
