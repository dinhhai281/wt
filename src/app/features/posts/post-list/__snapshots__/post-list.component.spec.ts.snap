// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <div
    class="grid grid-flow-row"
  >
    
    <article
      class="border-red-500 border-2 rounded-md bg-white p-4 mb-2 grid grid-cols-4 gap-2 cursor-pointer"
    >
      <div
        class="col-span-1"
      >
         1 1 
      </div>
      <div
        class="col-span-3"
      >
        <div
          class="text-lg text-pink-700 font-semibold"
        >
           title 
        </div>
        <div>
           body 
        </div>
      </div>
    </article>
  </div>
</div>
`;
